src/en/dynasty/src/eu/kanade/tachiyomi/extension/en/dynasty/Dynasty.kt

```diff
@@ -0,0 +1,698 @@
+            }
+        }
+
+        return if (data.type != "Doujin") {
+            chapterList.asReversed()
+        } else {
+            chapterList
+        }
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        return baseUrl + chapter.url
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val url = chapter.url
+            .removeSuffix("/")
+
+        assert(
+            url.count { it == '/' } == 2 &&
+                url.startsWith("/chapters/"),
+        ) {
+            "Refresh Chapter List"
+        }
+
+        return GET("$baseUrl${chapter.url}.json", headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val data = response.parseAs<ChapterResponse>()
+
+        return data.pages.mapIndexed { index, page ->
+            Page(index, imageUrl = baseUrl + page.url)
+        }
+    }
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = CHAPTER_FETCH_LIMIT_PREF
+            title = "Chapters Fetch Limit"
+            entries = CHAPTER_FETCH_LIMITS.map { "$it pages" }.toTypedArray()
+            entryValues = CHAPTER_FETCH_LIMITS
+            setDefaultValue(CHAPTER_FETCH_LIMITS[0])
+            summary = """
+                Limits how many pages of an entry are fetched for chapter list
+                Mostly applies to Doujins
+
+                More pages mean slower loading of chapter list
+
+                Currently fetching %s
+            """.trimIndent()
+        }.also(screen::addPreference)
+    }
+
+    private val SharedPreferences.chapterFetchLimit: Int
+        get() = getString(CHAPTER_FETCH_LIMIT_PREF, CHAPTER_FETCH_LIMITS[0])!!.let {
+            if (it == "all") {
+                Int.MAX_VALUE
+            } else {
+                it.toInt()
+            }
+        }
+
+    private val covers: Map<String, Map<String, String>> by lazy {
+        this::class.java
+            .getResourceAsStream("/assets/covers.json")!!
+            .bufferedReader().use { it.readText() }
+            .parseAs()
+    }
+
+    private fun getCoverUrl(directory: String?, permalink: String): String? {
+        directory ?: return null
+
+        if (directory == "chapters") {
+            return buildChapterCoverFetchUrl(permalink)
+        }
+
+        val file = covers[directory]?.get(permalink)
+            ?: return null
+
+        return buildCoverUrl(file)
+    }
+
+    private fun buildCoverUrl(file: String): String {
+        return baseUrl.toHttpUrl()
+            .newBuilder()
+            .addEncodedPathSegments(
+                file.removePrefix("/")
+                    .substringBefore("?"),
+            )
+            .fragment(COVER_URL_FRAGMENT)
+            .build()
+            .toString()
+    }
+
+    private fun buildChapterCoverFetchUrl(permalink: String): String {
+        return HttpUrl.Builder().apply {
+            scheme("http")
```

Comment:
Is it really not `https`?

Reply:
this is just a dummy url to be intercepted

Reply:
It would be beneficial to change it for static analysis purposes.